File: backend/app/schemas/state.py
from datetime import datetime

from pydantic import BaseModel

from app.schemas.equipment import EquipmentRead
from app.schemas.event import EventRead
from app.schemas.permit import PermitRead
from app.schemas.plant import PlantRead
from app.schemas.sensor import SensorRead
from app.schemas.worker import WorkerRead
from app.schemas.zone import ZoneRead


class ZoneState(BaseModel):
    """One zone with everything currently in it — the frontend's per-zone building block."""

    zone: ZoneRead
    workers: list[WorkerRead]
    equipment: list[EquipmentRead]
    sensors: list[SensorRead]
    active_permit_count: int


class PlantState(BaseModel):
    """Aggregate snapshot of the whole plant; what the frontend polls."""

    plant: PlantRead
    generated_at: datetime
    zones: list[ZoneState]
    active_permits: list[PermitRead]
    recent_events: list[EventRead]

File: src/features/plant/types/state.ts
import type { Equipment } from './equipment'
import type { PlantEvent } from './event'
import type { Permit } from './permit'
import type { PlantProfile } from './plant'
import type { Sensor } from './sensor'
import type { Worker } from './worker'
import type { Zone } from './zone'

export interface ZoneState {
  zone: Zone
  workers: Worker[]
  equipment: Equipment[]
  sensors: Sensor[]
  activePermitCount: number
}

/** The aggregate live snapshot the frontend polls — mirrors GET /api/v1/state. */
export interface PlantState {
  plant: PlantProfile
  generatedAt: string
  zones: ZoneState[]
  activePermits: Permit[]
  recentEvents: PlantEvent[]
}
